pkg/build/ignore/docker/dockign.go

```diff
@@ -0,0 +1,116 @@
+package docker
+
+import (
+	"bufio"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/golang/glog"
+	"github.com/openshift/source-to-image/pkg/api"
+)
+
+type DOCKIGN struct {
+	Config *api.Config
+}
+
+func New(req *api.Config) *DOCKIGN {
+	d := &DOCKIGN{
+		Config: req,
+	}
+	return d
+}
+
+func (b *DOCKIGN) Ignore(config *api.Config) {
+	path := filepath.Join(config.WorkingDir, "upload", "src", api.DefaultIgnoreFile)
```

Comment:
Yeah, these literals were used in clone.go a decent amount.  Part of my duplicating the pattern in dockign.go was to see if that was an oversight in prior reviews, or a toleration because it was in one file or something like that, where the review of this change would potentially flesh out the answer.

That said, the fact that filepath.Join, after the first argument, want to take single elements in a directory path (where the Join handles the separator) could also come into play here.  You would only build constants for "upload" and "src" and "tmp" (used in clone.go), not the combined paths.

I'll move on that approach, but can adjust if this particular comment thread progresses longer.